src/pybind/mgr/k8sevents/module.py

```diff
@@ -0,0 +1,1029 @@
+# Integrate with the kubernetes events API. 
+# This module sends events to Kubernetes, and also captures/tracks all events
+# in the rook-ceph namespace so kubernetes activity like pod restarts,
```

Comment:
At this point the code just applies the kubernetes config from the running cluster - but it could be enhanced to pull in a config file to load the kubernetes config to allow kubernetes events to be sent and retrieved from an external k8s cluster